cpu/stm32f1/periph/uart.c

```diff
@@ -136,9 +135,9 @@ int uart_init_blocking(uart_t uart, uint32_t baudrate)
     gpio_init(rx_pin, GPIO_DIR_IN, GPIO_NOPULL);
 
     /* configure UART to mode 8N1 with given baudrate */
-    divider = ((float)bus_freq) / (16 * baudrate);
-    mantissa = (uint16_t)floorf(divider);
-    fraction = (uint8_t)floorf((divider - mantissa) * 16);
+    bus_freq /= baudrate;
```

Comment:
Yes, look at the switch-case statement. If no proper UART is given it will return.